### PR TITLE
OpenStack regions restrictions

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
We need to have logic that provides a restriction to OpenStack regions. We need to filter out those regions that are not currently supported.

Changes proposed in this pull request:

- logic in the Provisioner that allows you to create an OpenStack cluster only in designated regions 
- unit tests update

### CODE REVIEW

File: resources/kcp/charts/provisioner/templates/deployment.yaml [hunk 1]
```diff
@@ -126,6 +126,8 @@ spec:
               value: {{ .Values.gardener.clusterCleanupTimeout | quote }}
             - name: APP_PROVISIONING_TIMEOUT_BINDINGS_CREATION
               value: {{ .Values.support.bindingsCreationTimeout | quote }}
+            - name: APP_OPENSTACKCONFIG_REGIONS
+              value: {{ .Values.openStackConfig.regions | squote }}
```
[the remark this answers is not in the document]
Comment (mvshao) [reply resources/kcp/charts/provisioner/templates/deployment.yaml:130]: Yes, that is intentional `squote` adds the single quote (needed for env value and JSON. `quoute` adds double quote

**Review by Disper — APPROVED** [review]

**Pull request merged**